### ISSUE TITLE
LE: Can't get automatic HTTPS to work on Google Container Engine (GKE)

### ISSUE THREAD

**Issue by admiralobvious** [mholt/caddy#407]
I have been trying to get automatic HTTPS to work on my Kubernetes cluster on GKE. I was starting caddy with `caddy -log stdout -agree -email myemail@email.com` and using the following Caddyfile:
```
alexferl.com {
    gzip
}

www.alexferl.com {
    redir https://alexferl.com
}
```
I get the following error:
```
2015/12/08 23:14:36 [INFO] acme: Obtaining bundled certificates for alexferl.com, www.alexferl.com
2015/12/08 23:14:36 [INFO] acme: Trying to solve HTTP-01
2015/12/08 23:14:38 [INFO] acme: Trying to solve TLS-SNI-01

2015/12/08 23:14:39 [alexferl.com] failed to get certificate: The server could not validate our request.
[www.alexferl.com] failed to get certificate: The server could not validate our request
```
It seems the LE server can't connect back to me, but I have a load-balancer with a static IP (104.196.35.65) pointing to the Kubernetes workers on ports 80 and 443 and DNS entries for alexferl.com and www.alexferl.com pointing to that static IP so in theory it should work.

```
$ host alexferl.com
alexferl.com has address 104.196.35.65
$ host www.alexferl.com
www.alexferl.com has address 104.196.35.65
```
I have also tried various different Caddyfile configs (just alexferl.com, just www.alexferl.com etc.) without success.

Any ideas?

**Comment by mholt** [comment]
Are you sure you can guarantee that the load balancer is always proxying the connection from the ACME server to the machine that's running Caddy totally transparently? SSL termination will break it the TLS-SNI-01 challenge, and proxying HTTP-01 insufficiently will cause that challenge to break, too. Given that there's nothing useful in the process log, I'm guessing the requests aren't even getting to your machine.

Until the DNS challenge is implemented by Let's Encrypt, you're on your own if you must be behind a load balancer other than Caddy. :-/

**Comment by admiralobvious** [comment]
It's using the network (L3) load-balancer so it is not terminating SSL and just forwards the whole TCP stream to the workers. I was thinking it would work, but it seems that it doesn't since I have found that people on AWS with ELB have the same issue so it's really a lets-encrypt "problem" and not the caddy implementation of lets-encrypt.

Oh well! Thank you!

**Comment by mholt** [comment]
Interesting - but, at least you are not alone! :smile: Best of luck.

**Comment by mholt** [comment]
@admiralobvious FWIW the DNS challenge will be available in Caddy 0.9, which means it should work well behind load balancers.

**Comment by admiralobvious** [comment]
@mholt Awesome! I'll give it a try once 0.9 is released. Thank you!
